avocado/utils/genio.py

```diff
@@ -174,3 +181,21 @@ def write_one_line(filename, line):
     :type line: str
     """
     write_file(filename, line.rstrip('\n') + '\n')
+
+
+def write_file_or_fail(filename, data):
+    """
+    Write to a file and raise exception on write failure
+
+    :param filename : Path to file
+    :type filename : str
+    :param data : Data to be written to file
+    :type data : str
+    :raises GenIOException : On write Failure
+    """
+    fd = os.open(filename, os.O_WRONLY)
+    try:
+        ret = os.write(fd, data)
+    except OSError as details:
+        raise GenIOException("The write to %s failed: %s" % (
+                             filename, details))
```

Comment:
Well we'd have to raise some exception as OSError does not (for reason I don't know) list the filename on failures. It is present, but it's not listed in the str output. Therefor I agree with raising an exception, I don't really care whether IOError/OSError or GenIOException (writing this I noticed the Exception vs. Error, I don't really care as the IDE will complete it automatically but the Error is more standard indeed).